[🦄 🦄 Uniswap Merge] 4.13.0

W3stside opened the pull request:
# Summary
Closes our commander Uniswap backmerge task

Updates us to [4.13.0](https://github.com/Uniswap/uniswap-interface/releases/tag/v4.13.0)

Desc. WIP

## Testing
- [ ] Everything!

## NOTE
- some styles broke (sorry @biocom): settings page is broken in dark mode

Commit:
ignore state/data/generated.ts

Commit:
fix: run graphql:codegen on yarn build (#1933)

* run graphql codegen on build

* yarn.lock

* remove ms.macro

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix: error messages from walletconnect provider (#1943)

moodysalem committed:
fix(position page): link to the token page of the explorer instead of the address page from the position page

Commit:
feat: handle chain id in subgraph api (#1938)

* build dynamic subgraph url based on chain id

* reset api state (query cache) on chain id change

* removed dependency on rtk-query/graphql

* add error message

Commit:
feat: add fee tier distribution badge (#1862)

* integrate with The Graph and auto-select fee tier

* restored

* addressed some design feedback

* add pulsing animation on feeAmount change

* simplify fee tier title

* adjust button radios

* addressed some design feedback

* log ReactGA events

* ignore data while fetching

* update to use new generated queries

* remove deleted file

* add usefeetierdistribution hook

* invalidate cache and standardize the outlined card

* added react ga

* fix show options logic

* address design feedback

* show % select in minified view

* updated merge error

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
feat(pools): fetch pool tick data from uniswap v3 subgraph (#1932)

* add hooks to fecth ticks data and active liquidity

* cleanup

* add polling interval

* moved ms.macro types to dev deps

* generate graphql schema on build

* added @types/ms.macro

* use clone deep

moodysalem committed:
refactor: logs hook (#1941)

* feat(logs): add an infrastructure for fetching logs in a declarative way

* use the logs hook in the vote page, first pass

* fix comment

* bit of cleanup

* unused imports

* improve loading indicator on vote page

* some testnet behavior improvements

* fix loader state

* loading state nit

* show correct indexes

* remove the unnecessary retry code

* first pass at the slice

* no throws

* loading indicator should go away if not connected

* use the logs slice for the logs hook state

* style changes per cal's request

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
various edits

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix: reduce call retries by specifying a block tag in multicall

ianlapham committed:
update defaults (#1955)

moodysalem committed:
fix: use a multicall that allows limiting gas of the individual calls (#1953)

* fix: use a multicall that allows limiting gas of the individual calls

* use the latest multicall addresses

moodysalem committed:
fix: use an arbitrum multicall that returns a better block number

Commit:
remove loading indicator from fee tier selection (#1954)

Commit:
fix: invalidate subgraph cache with provideTags (#1962)

* invalidate queries using tags

* enhance generated api with provide tags

* clean up

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix: do not reverse the proposals list in place

moodysalem committed:
fix(position image): increase the gas required constraint for the token URI

NoahZinsmeister committed:
no more `borderRadius` errors! (#1968)

NoahZinsmeister committed:
remove a lot of dead code (#1970)

moodysalem committed:
refactor(devx): show a warning when a call fails due to insufficient gas allowed

moodysalem committed:
chore: update yarn.lock

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

JFrankfurt committed:
feat(L2): network alert cards (optimism and arbitrum) (#1957)

* design review for arbitrum UI

* add optimism L2 UI

* better gradients, remove v2 links/routes, some minor padding and margin fixes

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

NoahZinsmeister committed:
fix l2 routing bug

NoahZinsmeister committed:
fix last commit

JFrankfurt committed:
fix(L2): routing fixes (#1980)

* Revert "fix last commit"

This reverts commit 9f5764aab31b7653f4046be00eae707460513d4f.

* Revert "fix l2 routing bug"

This reverts commit f6dea47907016a8ec820cf1b9b4a95e9f8e15be5.

* drops routing changes for L2

* pool v2 switch networks message

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
stabilize fee distribution in hook (#1979)

Commit:
fix: enforce styled-components/macro usage and upgrade react-scripts (#1981)

* upgrade react-scripts and enforce macro rule

* Fix code style issues with ESLint

Co-authored-by: Lint Action <lint-action@samuelmeuli.com>

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
improve liquidity depth hooks and data fetching (#1989)

Commit:
feat(pools): add liquidity chart range input primitives (#1990)

* first iteration of liquidity chart primitives

* add tickprocessed type

* clean up

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
chore: update the contribution guidelines (#1987)

* chore: update the contribution guidelines

* move

* note about non-deterministic steps

* little bit of separation

* language nits, add accessibility

* add reasoning

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

ChrisFox-Eth committed:
Updating two public-facing URLs to docs (#1992)

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

NoahZinsmeister committed:
feat: Add Optimism (#1923)

* preliminary support

* fix multicall abi

add explorer support

* Fix code style issues with ESLint

* fix gas estimation

* fix todo tests

* fix gas estimation comments

* anonymize links to kovan etherscan and poll more frequently on optimistic kovan

* separate pool creation

* remove supported chain id

* remove the blocktag

* do not use the blocktag on optimism only

* give more gas to tokenURI for optimism

* update sdk

* temp fix to the block tag thing (do not update the block number from the fetch block number)

* remove unused import

* gasRequired -> multi-network

* bump quoter gas limit to 6m on optimism

* move the gas required parameter by chain id one level up

* missed a hook

* retry fetching receipts for optimism as many times as arbitrum

* fix duplicate enum, add optimism as well to retry options config

* fix: state not getting updated after a transaction confirmation

* bump sdk version

* update for mainnet optimism (#1998)

* fix for calculateGasMargin on optimism

Co-authored-by: Lint Action <lint-action@samuelmeuli.com>
Co-authored-by: Moody Salem <moody.salem@gmail.com>

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: simplify liqudity depth hooks and improve chart primitives (#1999)

* updated hooks and chart primitives

* add lodash.inrange to package.json

Commit:
feat(pool): add liquidity distribution range slider and update add liquidity layout (#1829)

* first iteration of useTicks and useActiveLiquidity

* feat(pools): add liquidity depth chart (#1835)

* cleanup

* use area chart instead of bar chart

* check for undefined rather than falsy

* feat: range buttons based on fee amount (#1870)

* range buttons based on fee amount

* hardcode percentages as ticks

* increase blocksperfect

* feat: optimize add liquidity charts (#1880)

* ignore syncing state

* remove surrounding ticks

* avoid processing price1 as it is unused

* cleanup

* feat: add zoom buttons to liquidity depth chart (#1882)

* ignore syncing state

* remove surrounding ticks

* avoid processing price1 as it is unused

* cleanup

* first pass at +/- zoom buttons

* remove console.log, cleanup

* use real price for price line

* updated brush handles to latest spec

* added % labels to handles

* round tick to nearest usable tick

* first pass at brushable area chart with d3

* first pass at brushable area chart with d3

* rework

* address PR comments

* add brush handles

* address PR comments

* further improvements

* feat(pools): improve full range support + capital efficiency warning (#1903)

* handle min and max prices in add liquidity

* cleaned up

* use flag to denote full range

* reset inputs on fee select

* fixed merge conflict

* handle full range in positions preview

* fixed invalid range when tokens are reversed

* use formatTickData

* updated layout

* cleaned up layout

* fixed address

* avoid re-rendering deposit amounts

* added zoom behavior and more styling

* renamed chart

* renamed main chart file;

* add brush tooltips

* remove chart title

* added accents to brush handles

* more work

* moved to file

* modularize chart components

* fix maximum depth

* added brush labels

* cleanup

* cleanup

* set up zoom

* added new components

* improved brush and zoom integration

* cleaned up clip path

* fixed clip paths

* integrated with the graph changes

* adjust fee selector

* fix data error

* add bar chart

* polish

* merged

* clean up error

* cleaned up after merge

* visual improvements

* moved +/- buttons to the right/left

* removed margin bottom

* removed unsused

* fix brush labels % change

* use d3.axisBottom

* updated labels

* improve brush range

* fix one brush change only

* adjust zoom and clippath

* use bars

* use area

* adjust axis bottom to mocks;

* improved bars

* show bar colors

* better handle full range

* adjust colors for light mode

* updated to mocks

* adjusted handles for visibility

* switch to area

* add react ga events

* adjusted to mocks

* memo brush domain to avoid re-renders

* fix inputstepcounter color

* adjust handles

* rely on the graph sorting tickidx

* use curvestepafter

* updated polish

* merged main

* add clamping and other fixes

* highlight selected area using a mask

* use price instead of % for labels

* delete unused

* refine ux

* relayout

* improve hooks

* adjust layout for mobile

* fixed card color

* adjust padding

* preent tick overflow

* flip handles sooner

* delete bars.tsx

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

NoahZinsmeister committed:
update multicall address

Commit:
fix: min/max price inverted in positions page (#2002)

* fix param ordering in inverter hook

* use object param

moodysalem committed:
fix: retry calls that return header not found errors (#2004)

Commit:
fix: make liquidity chart and input step counter responsive (#2005)

* make chart responsive

* make input step counter text responsive

* clean up chart

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
chore(token lists): bump the token list spec version

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
support optimism and arbitrum subgraphs (#2001)

JFrankfurt committed:
fix(L2): polish (#1988)

* polish

move getINfoLinkByChainId to chain constants

pr review - translations

optimism token list, typo, text color for L1 switch

use <Trans /> instead of t

undo unintentional tokenlist order change

use {'Optimism'} instead of {name}

switch deadline implementation

remove unused TYPE import

switch to generalized CHAIN_INFO instead of L2_INFO so we don't have to check in components

add target chain id param to optimism bridge

fix a minor breakpoint issue

reduce sigfigs for header balance

update network card dropdown text for optimism

remove list code

refactor SupportedChainId

* SupportedL1ChainId | SupportedL2ChainId -> SupportedChainId

moodysalem committed:
fix: price tokens on optimism mainnet using DAI

moodysalem committed:
fix: remove dollar sign from translation of eth amount

fixes https://github.com/Uniswap/uniswap-interface/issues/1986

Commit:
fix: add liquidity flow polish (#2017)

* addressed feedback

* set initial, min and max zoom levels

* better handle 0

* avoid formatting range selector

* polish `not created` state

* remove unused import

ianlapham committed:
fix: update CMC priority within lists (#2019)

moodysalem committed:
fix(routing): updates routing constants for layer 2 networks to use ETH (#2018)

Commit:
fix: properly set upper/lower base/quote on manual inversion (#2023)

Commit:
load pool data for each tier to determine state (#2026)

Commit:
compare with undefined rather than falsy (#2027)

Commit:
fix: add liquidity ux polish (#2024)

* zoom out on intiial

* adjust initial zoom ranges

* remove unneeded reactga event

* adjust full range warning copy

* update zoom

* adjust zoom ranges and label around current price

moodysalem committed:
fix(L2): use 10 bips slippage for swaps by default on L2 (#2035)

* use 10bip slippage on L2

* update deps

Co-authored-by: Jordan Frankfurt <jordanwfrankfurt@gmail.com>

lukedonato committed:
add optional chaining for stakingInfo (#1950)

moodysalem committed:
fix(L2): enable optimism token list when switched to optimism (#2036)

* add title text and opacity variability to token lists

* add optimism token list

* show tokens from active lists

* sort up token lists
with tokens on the current chain

* fix up some type issues

prune out chainId changes

* clean up leftover any

* refactor token count mechanism

* handle plurals in title text string

* new combineMaps implementation

* remove custom plural

* address a couple nits

* show the number of tokens on current chain

Co-authored-by: Jordan Frankfurt <jordanwfrankfurt@gmail.com>
Co-authored-by: Justin Domingue <judo@uniswap.org>

Commit:
fix overflow on mobile (#2037)

JFrankfurt committed:
fix(L2): add downtime warning to add liquidity and adjust styles of network notification (#2034)

* add liquidity warning and network notification

* remove unneeded width

* use externalLink in LinkOutToBridge component

Co-authored-by: Moody Salem <moody.salem@gmail.com>

ianlapham committed:
fix: dismiss txn confirmation modal if on L2 (#2022)

* dismiss txn confirmation modal if on l2

* line up far right margin;

* remove submitted view on txns if on L2

* remove uneeded chainid

* wrap dismissal in useEffect

* update variable names

callil committed:
fix(L2): Design tweaks for L2 (Optimism) (#2038)

Style tweaks to header layout, network dropdown and pool pages

Co-authored-by: Jordan Frankfurt <jordanwfrankfurt@gmail.com>

ianlapham committed:
fix: support for isolated pool creation  (#2029)

* revert to remove ui polish changes

* left align blurb and fix bp issue

* fix translation wrapper

* remove preview screen for create, hide submitted txn UI for create, copy updates

* small fixes

* copy updates for create blurb

* add set price range title

* add translation

* fix some translations

Co-authored-by: Justin Domingue <judo@uniswap.org>

ianlapham committed:
update links (#2040)

Commit:
chore: support stubbing subgraph in integration tests (#2039)

* upgrade to 7.0

* first iteration of stubbing subgraph in integration tests

Commit:
fix: remove x-axis clamping behavior in liquidity chart (#2042)

* remove x-axis clamping behavior

* remove extra curly braces around errorMessage

Commit:
fix: revert remove x-axis clamping behavior in liquidity chart (#2045)

* Revert "fix: remove x-axis clamping behavior in liquidity chart (#2042)"

This reverts commit f86db00464448fe2c0e387aa01855f1710cd4fbd.

* removed extra curly braces around errorMessage

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix(arbitrum): use infura endpoints

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
refactor: clean up some constants

moodysalem committed:
fix: add DAI and WBTC and USDT to routing for Optimism

fixes https://github.com/Uniswap/uniswap-interface/issues/2016

moodysalem committed:
fix(block number): link to the correct page in etherscan for optimism block number links

ianlapham committed:
fix: update balance check in derived swap state (#2054)

* update balance check in derived swap state

* replace balance check with correct version of trade

Commit:
remove duplicate `set price range` title when there is liquidity (#2058)

Commit:
fix: add language selector in top-level menu (#2057)

* initial language selector

* set > icon opacity for consistency

* always reset menu to main on close

* always reset menu on menu open

moodysalem committed:
fix: all supported chain ids should not include the string keys of the enum

moodysalem committed:
chore: add one more unit test for ALL_SUPPORTED_CHAIN_IDS

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: remove chart clamping and show indicator when handle is off screen (#2064)

* initial off screen indicator

* adjust offscreen indicator

* add off screen handle indicator

* hide reset until we get a better behavior

* add svg.tsx

Commit:
fix: change panning cursor to col-resize (#2071)

* change panning cursor to col-resize

* revert to grabby, and support grabbing

Commit:
fix: set language using url param in language dropdown (#2063)

* set language using url param in language dropdown

* refactor common code into a hook

* add hook

* address pr feedback

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

ianlapham committed:
feat: only block on app url, mobile UI tweaks (#2073)

* start iframe ui updates

* replace hook with constant

* add origin check for blocked lists

* remove origin check for local unsupported list

* remove redundant retun

* remove iframe check

* undo local change

Commit:
remove unused dependencies (#2074)

Commit:
increase zoom out range (#2076)

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: invert selected range on RateToggle instead of clearing inputs (#2078)

brendanww committed:
Update WalletLink and @web3-react/walletlink-connector (#2079)

moodysalem committed:
fix(swap): double max gas for quoting a v3 swap for large trades

JFrankfurt committed:
set 5 min deadline on swaps and hide UI on L2 (#2081)

moodysalem committed:
chore: add blurb about release process (#2061)

* chore: add blurb about release process

* additional standard

* how to collect feedback

Commit:
feat: routing api integration (#2080)

* initial routing api integration

* add routing api slice

* display route in dialog

* addressed pr feedback

* switch to `get`

* renamed useRouterTradeExactIn to useRouter

* moving few files to later iteration

* removed unnecessary `as`

* switch to polling

* add todo for blocknumber freshness

* remove account-slippage-deadline

moodysalem committed:
chore: copy edit on CONTRIBUTING.md

moodysalem committed:
perf: list to token map function was doing too many reduces (#2095)

moodysalem committed:
perf: remove extra spread operators from the combine maps function

moodysalem committed:
feat: support connecting to a gnosis safe (#2096)

* feat: support connecting to a gnosis safe

* add the manifest.json attributes required by gnosis safe

* remove background fill

* copy

* only try injected provider if not active after trying safe

ianlapham committed:
add UniH to unsupported list (#2097)

Commit:
fix: better support decimal overflow (#2075)

* better support decimal overflow

* improve tests

* explicitly define num digits in toSignificant

* add tests for 0 decimals

* undo change to tryParseAmount and instead compute price without relying on parse

* remove console log and add test

* removed comment

* addressed pr feedback

* improve parsing logic

* addressed pr feedback

ianlapham committed:
update unsupported and broken lists (#2100)

moodysalem committed:
fix: remove html ids and unnecessary keys from the menu

moodysalem committed:
fix: prevent error if user selects ETH and WETH in add liquidity (#2112)

* fix: prevent error if user selects ETH and WETH in add liquidity

fixes https://github.com/Uniswap/uniswap-interface/issues/1763

* linting error

* revert changes to Tokens.ts

Commit:
test: set up component snapshot testing (#2102)

* set up snapshot testing

* improvements

* add tests for TextInput as an example

* Fix code style issues with ESLint

* add comment to custom-test-env file

* only set up needed providers

* include style rules in snapshots

* disable redux storage warning

* added setupTests to avoid boilerplate

Co-authored-by: Lint Action <lint-action@samuelmeuli.com>

NoahZinsmeister committed:
Revert "feat(routing): support mirror protocol routing as additional bases (#1375)"

This reverts commit 3198129af2eff07048eab2a9c041522495927e04.

NoahZinsmeister committed:
Revert "feat(uma): uma call option routing (#1385)"

This reverts commit 89d484d882e73ba6fe51a2cf66a893210cb0ce79.

ianlapham committed:
feat: use modal for txn confirmation on l2 (#2055)

* use modal for txn confirmation on l2

* update with error state

* remove 1 translation

* dedicated L2 component

* update styling on to be less jumpy

* add success animation to modal

* revert regular submitted content

* remove useless hook in popups

* remove import

Co-authored-by: Ian Lapham <ian_lapham@alumni.brown.edu>

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: toggle liquidity position when toggling rates (#2117)

If field A and field B are already populated, chances are the user wants
them to remain the same and only wants to invert the price range
selector.

The conversion isn't exact, but should be preferable to the previous
behavior.

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix: upgrade walletconnect to 1.5.1 (#2120)

* upgrade walletconnect to 1.5.0

* bump again

Commit:
document `useQueryCacheInvalidator`

explains why `chainId` is pulled directly from the store.

lint-action committed:
Fix code style issues with ESLint

JFrankfurt committed:
fix(L2): update downtime warning copy (#2114)

JFrankfurt committed:
add link (#2113)

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: add support for full range positions in add liquidity (#2090)

* remove arbitrary range buttons and move full range button

* better align full range to deposit

* support dragging range when in full range

* hack to support full range brushing

* restore zoom levels

* adjusted for mocks

* fix styling

* simplify type

* restore rate toggle change

* fix lower bound by looking at isSorted

* better align bottoms

* add reset button for full range positions

* only flip when not at limit

* fix +/- buttons in range selector

* add help link

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix: upgrade ethers to allow constructing transaction data compatible with eip-1559 (#2121)

* fix: upgrade ethers to allow constructing transaction data compatible with eip-1559

* bump the experimental provider

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: update mobile header to fix overflow (#2125)

Co-authored-by: Matthew Quinn <matthew.quinn@libertymutual.com>

Commit:
fix: center import pool text V2  (#2122)

* fix: center import pool text V2

* fix: updated based on feedback

Co-authored-by: Matthew Quinn <matthew.quinn@libertymutual.com>

Commit:
update full range documentation link

lint-action committed:
Fix code style issues with ESLint

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

JFrankfurt committed:
fix: optimism link and some copy changes (#2126)

* fix: optimism link

* fix: (un)planned downtime copy changes

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

dependabot[bot] committed:
chore(deps): bump tar from 6.1.0 to 6.1.4 (#2132)

Bumps [tar](https://github.com/npm/node-tar) from 6.1.0 to 6.1.4.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v6.1.0...v6.1.4)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
upgrade to walletconnect 1.5.2 (#2134)

Commit:
fix: rename weth9 to weth (#2133)

* update @uniswap/sdk-core and @uniswap/v3-sdk

* reorder

ianlapham committed:
update unsupported list (#2143)

Co-authored-by: Ian Lapham <ianlapham@Ians-MacBook-Pro.local>

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

Commit:
fix: avoid running BestV3Trade when tokens are the same (#2172)

* avoid running BestV3Trade when tokens are the same

* flipped condition

Commit:
add ReactGA event for full range (#2181)

Commit:
update chinese simplified and traditional display names (#2180)

moodysalem committed:
fix: proposal descriptions should be pulled from the right governor (#2176)

moodysalem committed:
fix: `#isTradeBetter` should check that output currencies are equal (#2177)

* fix: isTradeBetter should check that input/output are equal

* fix the fix

ianlapham committed:
update blocked address list (#2182)

Co-authored-by: Ian Lapham <ianlapham@Ians-MacBook-Pro.local>

moodysalem committed:
fix: do not construct routes with the same pool twice (#2196)

moodysalem committed:
chore: set `default_bump` to false per https://github.com/mathieudutour/github-tag-action/issues/85

NoahZinsmeister committed:
fix: use `@walletconnect/ethereum-provider` instead of `@walletconnect/web3-provider` (#2175)

* use @walletconnect/ethereum-provider

* bump to support client-side disconnect

zzmp committed:
chore(i18n): touch en-US to prevent lingui from adding a creation date (#2210)

Fixes #2198. Touching en-US.po before extracting messages prevents lingui from adding a creation date, maintaining deterministic builds and IPFS hashes.

zzmp committed:
chore(i18n): do not compile i18n messages (#2211)

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

zzmp committed:
chore: fix SEO and accessibility nits for Lighthouse (#2212)

* refactor: index.html

* chore: preconnect to GA

* chore(SEO): add meta content

* chore(SEO): add alt to ethereum logo

* chore(accessibility): add aria-labels to menus

* chore(accessibility): mark AppBody as main

* chore(accessibility): update nav link ids

* chore(accessibility): set html.lang to match i18n locale

* chore(refactor): mv html.lang to useEffect

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

moodysalem committed:
fix: update walletconnect for deep links and scalability (#2215)

zzmp committed:
chore: change language icon to globe (#2222)

* chore: change language icon to globe

zzmp committed:
chore(accessibility): increase contrast for connections (#2223)

NoahZinsmeister committed:
fix(governance): hot fix for `parseLog` crash

NoahZinsmeister committed:
fix: proposal formatting

zzmp committed:
chore: rm unused styled-system (#2234)

zzmp committed:
chore: update multicodec usage (#2233)

zzmp committed:
chore: remove unused cross-env (#2232)

zzmp committed:
chore: rm redux-devtools-extension (#2230)

redux-devtools-extension integration is already provided through @reduxjs/toolkit.

moodysalem committed:
fix: walletconnect on L2 issues with Rainbow (#2242)

* fix: remove the walletconnect bridge url

* fix: use the latest ethereum provider which fixes the optimism rainbow issue

moodysalem committed:
fix: turn service workers back on (#1944)

reconfigure service workers to only cache used assets (excluding .po language files and non-.var.woff2 font files)

* fix: turn service workers back on

* chore: configure service worker caches

* chore: add newline

* Fix code style issues with ESLint

* chore: limit service-worker caching

Co-authored-by: Zach Pomerantz <zzmp@uniswap.org>
Co-authored-by: Lint Action <lint-action@samuelmeuli.com>

Commit:
test: support stubbing subgraph data in cypress (#2259)

* upgrade to 7.0

* first iteration of stubbing subgraph in integration tests

* added fixtures

* add tests for fee tier distribution and liquidity chart

* remove unused test utils

* update yarn.lock

* fixed merge artifacts

ianlapham committed:
update token list (#2270)

zzmp committed:
fix(governance): special case bravo parsing (#2247)

* fix(governance): special case bravo parsing

* fix(governance): explicitly parse U+2018, U+2026

* chore: upgrade ethers to ^5.4.6

Updates parseLog to delay parsing nested properties, so that they will only fail when accessed.

* fix(vote): recover from invalid UTF-8 in proposal descriptions

* fix(vote): special case bravo proposal newlines

* chore: rm dead reference

JFrankfurt committed:
add arbitrum token list (#2268)

zzmp committed:
Revert "test: support stubbing subgraph data in cypress (#2259)" (#2269)

This reverts commit 00403a481034ad4a23fc454a818b5687237c246d.

zzmp committed:
fix: disable serviceworker in integration tests (#2274)

JFrankfurt committed:
update explorer links for arbitrum (#2279)

dependabot[bot] committed:
chore(deps): bump tar from 6.1.4 to 6.1.11 (#2280)

Bumps [tar](https://github.com/npm/node-tar) from 6.1.4 to 6.1.11.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v6.1.4...v6.1.11)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
update optimism subgraph url (#2282)

circle-hotaru committed:
fix: older browser suppport (#2228)

* fix: Object.fromEntries polyfill

* fix: update polyfills

* fix: update polyfills

zzmp committed:
chore: upgrade type system (#2276)

* chore: upgrade type system

Upgrade types, typescript, and formatters.

* fix: ListsState tests

JFrankfurt committed:
use network-specific help center link (#2285)

zzmp committed:
style: enforce object-shorthand (#2286)

zzmp committed:
fix: make non-interactive header elements unselectable (#2288)

zzmp committed:
fix: always paint StyledPollingNumber (#2254)

Adds a en-space to StyledPollingNumber so it is always painted. Fixes a layout shift when rendering the first block number.
StyledPollingNumber is taller than its container, so waiting to paint it causes a layout shift when rendering.

zzmp committed:
chore: paint currency selector on initialization (#2245)

Avoids a flash of "Select a token" as the App initializes.

NoahZinsmeister committed:
feat: governor bravo support (#2271)

* first pass

* Fix code style issues with ESLint

* address comments

Co-authored-by: Lint Action <lint-action@samuelmeuli.com>

zzmp committed:
chore: use native flatMap (#2231)

crowdin-bot committed:
chore(i18n): synchronize translations from crowdin [skip ci]

NoahZinsmeister committed:
Squashed commit of the following:

commit 5f1d1af62bcd47286aafacc18788f4e7e22dd7c0
Author: Aseem Sood <asood123@yahoo.com>
Date:   Fri Sep 3 12:12:39 2021 -0400

    update readme

    tweak

commit 80a5b95c0e0ae8934b5591c982077eaa813db747
Author: Noah Zinsmeister <noahwz@gmail.com>
Date:   Fri Sep 3 11:52:56 2021 -0400

    consolidate and standardize unsupported tokens

W3stside committed:
Merge remote-tracking branch 'uni-origin/main' into uni-merge/4.13.0

W3stside committed:
yarn lock

W3stside committed:
merge type/style/conflict fixes

W3stside committed:
Merge remote-tracking branch 'origin/develop' into uni-merge/4.13.0

biocom commented:
Is it possible to get a PR build on this?

W3stside committed:
fix supported networks support

W3stside committed:
fix settings

W3stside committed:
test snapshots

W3stside committed:
don't ignore generated

biocom commented:
Just for reference, some style items which need to be addressed separately from this PR:
<img width="226" alt="Screen Shot 2021-09-09 at 14 36 50" src="https://user-images.githubusercontent.com/31534717/132687051-76386921-d349-494a-a31c-9630a552f230.png">
<img width="308" alt="Screen Shot 2021-09-09 at 14 34 58" src="https://user-images.githubusercontent.com/31534717/132687053-bf0f2dc7-36a3-41d7-bb87-14fe614bc99a.png">
<img width="243" alt="Screen Shot 2021-09-09 at 14 34 33" src="https://user-images.githubusercontent.com/31534717/132687057-b58d435c-3ee8-462a-a1aa-3b45787b97b5.png">
<img width="162" alt="Screen Shot 2021-09-09 at 14 34 10" src="https://user-images.githubusercontent.com/31534717/132687060-5aa4cd8d-5d4a-4d37-8f9f-b70114edae21.png">
<img width="513" alt="Screen Shot 2021-09-09 at 14 33 56" src="https://user-images.githubusercontent.com/31534717/132687062-00ed6e10-8d3a-4ab5-afd6-552065fb095b.png">

W3stside committed:
move workflows to uni

W3stside commented on .eslintrc.json:
```diff
@@ -28,6 +28,20 @@
+    "@typescript-eslint/explicit-module-boundary-types": "off",
+    "react/react-in-jsx-scope": "off",
+    "object-shorthand": ["error", "always"],
+    "no-restricted-imports": [
```
we have to import `styled-components/macro` now

biocom replied:
Wasn't this the case already since previous UNI merge?

W3stside commented on package.json:
```diff
@@ -7,7 +7,12 @@
-    "@ethersproject/experimental": "^5.2.0",
+    "@ethersproject/experimental": "^5.4.0",
+    "@gnosis.pm/safe-apps-web3-react": "^0.6.0",
+    "@graphql-codegen/cli": "1.21.5",
```
lots of stuff we dont need...

alfetopito replied:
Shouldn't then we just not add them as dependencies?

W3stside replied:
but a lot of files depend on these packages existing in compile time so they need to remain lest we remove the files importing them

W3stside commented on package.json:
```diff
@@ -48,40 +57,45 @@
-    "cypress": "6.7.1",
+    "jest-styled-components": "^7.0.5",
+    "cypress": "^7.7.0",
+    "d3": "^7.0.0",
```
they use this in a page we dont use.

W3stside commented on package.json:
```diff
@@ -48,40 +57,45 @@
-    "ethers": "^5.2.0",
+    "ethers": "^5.4.6",
+    "graphql": "^15.5.0",
+    "graphql-request": "^3.4.0",
```
graphql also part of a page we dont use

W3stside commented on package.json:
```diff
@@ -104,20 +119,20 @@
+    "workbox-routing": "^6.1.0"
+  },
+  "resolutions": {
+    "@walletconnect/ethereum-provider": "1.6.4"
```
this is new. we need to check why

W3stside commented on package.json:
```diff
@@ -137,9 +152,11 @@
     "i18n:extract": "lingui extract --locale en-US",
-    "i18n:compile": "lingui compile"
+    "prei18n:extract": "touch src/locales/en-US.po",
+    "prestart": "yarn graphql:generate && yarn prei18n:extract"
```
new scripts to compile their graphql stuff

W3stside commented on src/components/Button/index.tsx:
```diff
@@ -153,63 +144,41 @@ export const ButtonSecondary = styled(Base)`
-  background-color: ${({ theme }) => theme.primary1};
-  color: white;
-
+export const ButtonOutlined = styled(Base)`
```
@biocom i would check the implication/implementation of this.

W3stside commented on src/components/Header/NetworkCard.tsx:
```diff
@@ -0,0 +1,235 @@
+import { Trans } from '@lingui/macro'
```
@biocom this entire comp is new and affecting the chainId visualisation

biocom replied:
OK, just visual changes right (that's what I've seen thus far)?

W3stside commented on src/components/Header/NetworkCard.tsx:
```diff
@@ -0,0 +1,235 @@
+    if (!library?.provider?.request || !chainId || !library?.provider?.isMetaMask) {
+      return
+    }
+    switchToNetwork({ library, chainId })
```
essentially looks like the code i added, haven't gotten it to work  tho

W3stside committed:
path fix

W3stside committed:
path fix / walletmodal

alfetopito commented on CONTRIBUTING.md:
```diff
@@ -33,15 +33,13 @@ Start with issues with the label
 <!-- # Translations
-TODO: read and re-enable
-Help CowSwap reach a global audience! 
+Uniswap uses [Crowdin](https://crowdin.com/project/uniswap-interface) for managing translations. 
+[This workflow](./.github/workflows/crowdin.yaml) uploads new strings for translation to the Crowdin project whenever code using the [lingui translation macros](https://lingui.js.org/ref/macro.html) is merged into `main`.
```
Should probably revert the changes in this file

alfetopito commented on cypress.json:
```diff
@@ -1,4 +1,5 @@
 {
+  "projectId": "yp82ef",
```
Should we create a project id for ourselves?

alfetopito commented on src/components/AccountDetails/Transaction.tsx:
```diff
@@ -1,9 +1,8 @@
-import React from 'react'
```
It's a tsx file, this is required, no?

BTW, I think this whole file should be reverted, these changes are necessary for the mod files AFAICT

alfetopito commented on src/components/AccountDetails/index.tsx:
```diff
@@ -1,16 +1,15 @@
+import styled, { ThemeContext } from 'styled-components/macro'
+import { SUPPORTED_WALLETS } from '../../constants/wallet'
 import { useActiveWeb3React } from '../../hooks/web3'
 import { clearAllTransactions } from '../../state/transactions/actions'
-import { shortenAddress } from 'utils'
-import { ExplorerDataType, getExplorerLink } from 'utils/getExplorerLink'
+import { shortenAddress } from '../../utils'
+import { ExplorerDataType, getExplorerLink } from '../../utils/getExplorerLink'
```
Keep absolute imports?

alfetopito commented on src/components/AddressInputPanel/index.tsx:
```diff
@@ -1,8 +1,9 @@
+import styled, { ThemeContext } from 'styled-components/macro'
 import useENS from '../../hooks/useENS'
 import { useActiveWeb3React } from '../../hooks/web3'
-import { ExternalLink, TYPE } from 'theme'
+import { ExternalLink, TYPE } from '../../theme'
```
Absolute imports

alfetopito commented on src/components/Badge/RangeBadge.tsx:
```diff
@@ -1,5 +1,3 @@
-import React from 'react'
```
I'm seeing this in several files, I'll stop adding comments regarding it.
Is this no longer required for tsx files?

alfetopito replied:
Maybe not, since 1 year ago https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

Was our version that old? :thinking:

alfetopito commented on src/components/Badge/index.tsx:
```diff
@@ -13,7 +13,7 @@ export enum BadgeVariant {
   WARNING_OUTLINE = 'WARNING_OUTLINE',
 }
 
-export interface BadgeProps {
```
Keep export?

alfetopito commented on src/components/CurrencyInputPanel/index.tsx:
```diff
@@ -17,7 +17,7 @@ import { Trans } from '@lingui/macro'
 import { Lock } from 'react-feather'
 import { AutoColumn } from 'components/Column'
-import { FiatValue } from 'components/CurrencyInputPanel/FiatValue'
+import { FiatValue } from './FiatValue'
```
Absolute imports

alfetopito commented on src/components/CurrencyLogo/index.tsx:
```diff
@@ -1,13 +1,13 @@
 
 export const getTokenLogoURL = (address: string) =>
-  `https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/${address}/logo.png`
+  `https://raw.githubusercontent.com/uniswap/assets/master/blockchains/ethereum/assets/${address}/logo.png`
```
Wow, they have their own fork of trust wallet assets repo now!

alfetopito commented on src/components/DoubleLogo/index.tsx:
```diff
@@ -1,7 +1,6 @@
-import React from 'react'
 import styled from 'styled-components/macro'
-import CurrencyLogo from 'components/CurrencyLogo'
+import CurrencyLogo from '../CurrencyLogo'
```
Absolute import

alfetopito commented on src/components/Logo/index.tsx:
```diff
@@ -1,11 +1,11 @@
 
 const BAD_SRCS: { [tokenAddress: string]: true } = {}
 
-export interface LogoProps extends Pick<ImageProps, 'style' | 'alt' | 'className'> {
```
Keep the export

alfetopito commented on src/components/Menu/index.tsx:
```diff
@@ -1,16 +1,22 @@
 import { Trans } from '@lingui/macro'
-
-import { ExternalLink } from 'theme'
+import { ExternalLink } from '../../theme'
```
Absolute import

alfetopito reviewed (commented):
62 files checked, ~400 to go

alfetopito commented on src/components/Popups/PopupItem.tsx:
```diff
@@ -1,13 +1,13 @@
 import { useRemovePopup } from '../../state/application/hooks'
 import TransactionPopup from './TransactionPopup'
 
-export const StyledClose = styled(X)`
```
Keep the exports

alfetopito commented on src/components/PositionList/index.tsx:
```diff
@@ -34,7 +34,7 @@ const MobileHeader = styled.div`
 `
 
-export type PositionListProps = React.PropsWithChildren<{
+type PositionListProps = React.PropsWithChildren<{
```
Keep the export?

alfetopito commented on src/components/PositionListItem/index.tsx:
```diff
@@ -1,29 +1,34 @@
+import { RowBetween } from 'components/Row'
 import HoverInlineText from 'components/HoverInlineText'
-import { DAI, USDC, USDT, WBTC, WETH9_EXTENDED } from 'constants/tokens'
+import { DAI, USDC, USDT, WBTC, WETH9_EXTENDED } from '../../constants/tokens'
```
absolute imports

alfetopito commented on src/components/PositionListItem/index.tsx:
```diff
@@ -117,7 +126,7 @@ const DataText = styled.div`
   `};
 `
 
-export interface PositionListItemProps {
```
keep exports

alfetopito commented on src/components/SearchModal/CurrencyList.tsx:
```diff
@@ -16,8 +16,8 @@ import CurrencyLogo from 'components/CurrencyLogo'
-import { isTokenOnList } from 'utils'
-import ImportRow from 'components/SearchModal/ImportRow'
+import { isTokenOnList } from '../../utils'
+import ImportRow from './ImportRow'
```
imports

alfetopito commented on src/components/SearchModal/CurrencySearch.tsx:
```diff
@@ -1,10 +1,10 @@
 import { FixedSizeList } from 'react-window'
 import { Text } from 'rebass'
-import { GpEther as ExtendedEther } from 'constants/tokens'
+import { ExtendedEther } from '../../constants/tokens'
```
imports

alfetopito commented on src/components/SearchModal/Manage.tsx:
```diff
@@ -1,14 +1,14 @@
-import { ManageLists } from 'components/SearchModal/ManageLists'
-import ManageTokens from 'components/SearchModal/ManageTokens'
+import { ManageLists } from './ManageLists'
+import ManageTokens from './ManageTokens'
```
imports

alfetopito commented on src/components/SearchModal/ManageLists.tsx:
```diff
@@ -1,36 +1,34 @@
-import { ExternalLink, LinkStyledButton, TYPE, IconWrapper } from '../../theme'
+import { acceptListUpdate, disableList, enableList, removeList } from '../../state/lists/actions'
+import { useActiveListUrls, useAllLists, useIsListActive } from '../../state/lists/hooks'
+import { ExternalLink, IconWrapper, LinkStyledButton, TYPE } from '../../theme'
```
imports

alfetopito commented on src/components/SearchModal/styleds.tsx:
```diff
@@ -1,43 +1,6 @@
-  text-align: left;
-  top: 80px;
-`
+import { RowBetween } from '../Row'
```
imports

alfetopito commented on src/components/Toggle/ListToggle.tsx:
```diff
@@ -31,7 +30,7 @@ const StatusText = styled(TYPE.main)<{ isActive?: boolean }>`
 `
 
-export interface ToggleProps {
+interface ToggleProps {
```
Exports

alfetopito reviewed (commented):
Reviewed files 63 to 110

alfetopito commented on src/components/Tooltip/index.tsx:
```diff
@@ -21,7 +21,7 @@ export default function Tooltip({ text, ...rest }: TooltipProps) {
   return <Popover content={<TooltipContainer>{text}</TooltipContainer>} {...rest} />
 }
 
-export function TooltipContent({ content, ...rest }: TooltipContentProps) {
```
export

alfetopito commented on src/components/TransactionConfirmationModal/index.tsx:
```diff
@@ -1,20 +1,24 @@
 import Circle from '../../assets/images/blue-loader.svg'
 import MetaMaskLogo from '../../assets/images/metamask.png'
-import { useActiveWeb3React } from 'hooks/web3'
+import { useActiveWeb3React } from '../../hooks/web3'
```
import

alfetopito commented on src/components/TransactionConfirmationModal/index.tsx:
```diff
@@ -39,7 +43,7 @@ const StyledLogo = styled.img`
   margin-left: 6px;
 `
 
-export function ConfirmationPendingContent({
```
export

alfetopito commented on src/components/TransactionSettings/index.tsx:
```diff
@@ -87,11 +89,12 @@ const SlippageEmojiContainer = styled.span`
 `
 
-export interface TransactionSettingsProps {
+interface TransactionSettingsProps {
```
export

alfetopito commented on src/components/WalletModal/Option.tsx:
```diff
@@ -1,6 +1,5 @@
-import React from 'react'
 import styled from 'styled-components/macro'
-import { ExternalLink } from 'theme'
+import { ExternalLink } from '../../theme'
```
import

alfetopito commented on src/components/WalletModal/index.tsx:
```diff
@@ -2,20 +2,20 @@ import { AbstractConnector } from '@web3-react/abstract-connector'
-import { ExternalLink, TYPE } from 'theme'
-import AccountDetails from '@src/components/AccountDetails'
+import { ExternalLink, TYPE } from '../../theme'
+import AccountDetails from '../AccountDetails'
```
imports

alfetopito commented on src/components/swap/ConfirmSwapModal.tsx:
```diff
@@ -2,11 +2,11 @@ import { Trans } from '@lingui/macro'
   ConfirmationModalContent,
   TransactionErrorContent,
-} from 'components/TransactionConfirmationModal'
+} from '../TransactionConfirmationModal'
```
import

alfetopito commented on src/components/swap/SwapModalFooter.tsx:
```diff
@@ -3,9 +3,9 @@ import { Currency, TradeType } from '@uniswap/sdk-core'
+import { ReactNode } from 'react'
 import { Text } from 'rebass'
-import { ButtonError } from 'components/Button'
+import { ButtonError } from '../Button'
```
import

alfetopito commented on src/components/swap/SwapModalHeader.tsx:
```diff
@@ -1,18 +1,18 @@
-import { FiatValue } from 'components/CurrencyInputPanel/FiatValue'
-import CurrencyLogo from 'components/CurrencyLogo'
+import { FiatValue } from '../CurrencyInputPanel/FiatValue'
+import CurrencyLogo from '../CurrencyLogo'
```
imports and exports

alfetopito commented on src/components/swap/SwapRoute.tsx:
```diff
@@ -1,11 +1,11 @@
-import { ThemeContext } from 'styled-components'
-import { TYPE } from 'theme'
+import { ThemeContext } from 'styled-components/macro'
+import { TYPE } from '../../theme'
```
import

alfetopito commented on src/connectors/index.ts:
```diff
@@ -52,15 +41,15 @@ export function getNetworkLibrary(): Web3Provider {
+  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
 })
 
+export const gnosisSafe = new SafeAppConnector()
```
:thinking:

alfetopito commented on src/constants/routing.ts:
```diff
@@ -4,22 +4,21 @@ import { SupportedChainId } from './chains'
   AMPL,
   DAI,
-  GpEther as ExtendedEther,
+  ExtendedEther,
```
imports

alfetopito reviewed (commented):
Reviewed files 111 to 155.
Starting `custom` files next

alfetopito commented on src/custom/constants/chains/chainsMod.ts:
```diff
@@ -19,3 +21,11 @@ export const NETWORK_LABELS: { [chainId in SupportedChainId | number]: string }
+  SupportedChainId.ROPSTEN,
+  SupportedChainId.RINKEBY,
+  SupportedChainId.GOERLI,
+  SupportedChainId.KOVAN,
```
We should probably disable ropsten, goerli and kovan

W3stside replied:
added xdai as well, thanks

alfetopito commented on src/custom/pages/Swap/SwapMod.tsx:
```diff
@@ -878,19 +873,15 @@ export default function Swap({
       {!swapIsUnsupported ? null : !isSupportedWallet ? (
         <UnsupportedCurrencyFooter
-          show={!isSupportedWallet}
+          show={swapIsUnsupported}
```
Guess you missed the `!isSupportedWallet` flag here

W3stside replied:
@alfetopito don't get this comment, did i make a mistake or

alfetopito commented on src/hooks/useBestV3Trade.ts:
```diff
@@ -1,10 +1,12 @@
 import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
 import { encodeRouteToPath, Route, Trade } from '@uniswap/v3-sdk'
+import { SupportedChainId } from '@src/constants/chains'
```
Path?

alfetopito reviewed (commented):
Reviewed files 156 to 300

alfetopito commented on src/theme/components.tsx:
```diff
@@ -275,7 +220,7 @@ const rotate = keyframes`
 `
 
-export const Spinner = styled.img`
+const Spinner = styled.img`
```
export?

alfetopito commented on src/utils/isTradeBetter.ts:
```diff
@@ -16,7 +16,7 @@ export function isTradeBetter(
     tradeA.tradeType !== tradeB.tradeType ||
     !tradeA.inputAmount.currency.equals(tradeB.inputAmount.currency) ||
-    !tradeB.outputAmount.currency.equals(tradeB.outputAmount.currency)
+    !tradeA.outputAmount.currency.equals(tradeB.outputAmount.currency)
```
Opsie

anxolin replied:
🔥

alfetopito commented on src/utils/retry.ts:
```diff
@@ -9,7 +9,7 @@ function waitRandom(min: number, max: number): Promise<void> {
  * This error is thrown if the function is cancelled before completing
  */
-export class CancelledError extends Error {
+class CancelledError extends Error {
```
Export?

alfetopito commented on src/utils/unwrappedToken.ts:
```diff
@@ -1,5 +1,5 @@
 import { Currency } from '@uniswap/sdk-core'
-import { GpEther as ExtendedEther, WETH9_EXTENDED } from 'constants/tokens'
+import { ExtendedEther, WETH9_EXTENDED } from '../constants/tokens'
```
Import?

alfetopito reviewed (commented):
Reviewed 301 to 457

...

No time to try out the app

MareenG commented:
There is also a small style issue in expert mode when adding a recipient
<img width="722" alt="Bildschirmfoto 2021-09-10 um 09 32 13" src="https://user-images.githubusercontent.com/34510341/132817391-d48d6396-d99b-4a5b-baa2-0a8dbc5984a6.png">

anxolin commented on src/components/SearchModal/CurrencySearch.tsx:
```diff
@@ -203,14 +207,15 @@ export function CurrencySearch({
               <CurrencyList
                 height={height}
-                currencies={filteredSortedTokensWithETH}
+                currencies={disableNonToken ? filteredSortedTokens : filteredSortedTokensWithETH}
```
@W3stside nothing from this PR, but I think we can take a look to one issue we had, where we don't show wxDAI and WETH on top of the search token popopup:
![image](https://user-images.githubusercontent.com/2352112/132844296-5216155b-b8b7-4650-91a5-2cc13d41a928.png)


I think is probably here where we need to touch

W3stside replied:
we have a modded version that should be working properly, maybe importis broken

anxolin commented on src/components/SwitchLocaleLink/index.tsx:
```diff
@@ -17,46 +14,35 @@ const Container = styled(TYPE.small)`
+  return (
+    <Container>
+      <Trans>
+        Uniswap available in:{' '}
```
@nenadV91 @W3stside , this is locale related, and also has "Uniswap" in it, so we will need to personalize

W3stside replied:
it is already modded

anxolin commented on src/components/TransactionConfirmationModal/index.tsx:
```diff
@@ -39,7 +43,7 @@ const StyledLogo = styled.img`
 `
 
-export function ConfirmationPendingContent({
+function ConfirmationPendingContent({
```
@biocom @W3stside , not sure if this is what I think it is, but it looks like something we want and Michel was going to work in. Right now there/s no modal confirmation when you approve/wrap/unwrap, so we would need one. 

Is this it?

anxolin commented on src/constants/addresses.ts:
```diff
@@ -5,50 +5,64 @@ import { SupportedChainId } from './chains'
-  ...constructSameAddressMap('0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696', false),
-  [SupportedChainId.ARBITRUM_KOVAN]: '0xc80e33a6f02cf08557a0ca3d94d1474d73f64bc1',
-  [SupportedChainId.ARBITRUM_ONE]: '0x021CeAC7e681dBCE9b5039d2535ED97590eB395c',
+export const UNI_ADDRESS: AddressMap = constructSameAddressMap('0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984')
```
@W3stside there's a lot of changes here, and i think we override many of this addresses to add xDAI, could you review it more indetail for another PR?

anxolin commented on src/utils/getLibrary.ts:
```diff
@@ -1,4 +1,13 @@
+import ms from 'ms.macro'
+import { SupportedChainId } from '../constants/chains'
+
+const NETWORK_POLLING_INTERVALS: { [chainId: number]: number } = {
```
Should we use this and add xDAI?

anxolin commented on src/utils/anonymizeLink.test.ts:
```diff
@@ -0,0 +1,41 @@
+import { anonymizeLink } from './anonymizeLink'
+
+describe('#anonymizeLink', () => {
+  it('does nothing to non-urls', () => {
```
@alfetopito @W3stside this is interesting. Probably important for legal reasons. We should try not to leak these

anxolin reviewed (approved):
Awesome David!

What a huge PR. Let's try to do merges every week to make this less painful for all of us. 

Approving, however, I think we should do followup PRs and don't merge to DEVELOP for now, unless we think this is all save to merge and the release that @W3stside is making to do today, has been already branched off.

elena-zh commented:
To add one more issue to the UI issues list in this PR: 
scroll appears when open "Manage tokens lists" modal
![image](https://user-images.githubusercontent.com/70885163/132868953-bfaaea94-3fd1-4a17-bc9a-9d1793228c49.png)

MareenG commented:
Another small design issue 
<img width="488" alt="Bildschirmfoto 2021-09-13 um 14 36 31" src="https://user-images.githubusercontent.com/34510341/133084720-2ab6b37b-14ca-4f20-b68c-6babad10ba5c.png">

MareenG commented:
In darkmode the contrast is a bit better so see the overlapping 
<img width="452" alt="Bildschirmfoto 2021-09-13 um 14 40 23" src="https://user-images.githubusercontent.com/34510341/133085240-ebd9f1a2-fafb-4e0b-ac34-c4dd3cc89032.png">

elena-zh commented:
> In darkmode the contrast is a bit better so see the overlapping

It is reported in #1354

W3stside committed:
Merge remote-tracking branch 'origin/develop' into uni-merge/4.13.0

W3stside committed:
Merge branch 'develop' into uni-merge/4.13.0

lint-action committed:
Fix code style issues with Prettier

W3stside committed:
Merge remote-tracking branch 'origin/develop' into uni-merge/4.13.0

W3stside committed:
small fixes post merge

W3stside committed:
Merge branch 'uni-merge/4.13.0' of github.com:gnosis/gp-swap-ui into uni-merge/4.13.0

lint-action committed:
Fix code style issues with Prettier

W3stside commented on src/components/FeeSelector/index.tsx:
```diff
@@ -1,81 +1,228 @@
+  [FeeAmount.HIGH]: {
+    label: '1',
+    description: <Trans>Best for exotic pairs.</Trans>,
+  },
```
right now no as it's for pools

W3stside committed:
pr comments

1. imports/exports

W3stside committed:
import path

W3stside committed:
Merge branch 'uni-merge/4.13.0' of github.com:gnosis/gp-swap-ui into uni-merge/4.13.0

W3stside committed:
pr comments

W3stside commented on src/state/lists/hooks.ts:
```diff
@@ -12,30 +13,30 @@ export type TokenAddressMap = Readonly<{
 }>
 
+type Mutable<T> = {
+  -readonly [P in keyof T]: Mutable<T[P]>
```
readonly

W3stside commented on src/hooks/useERC20Permit.ts:
```diff
@@ -114,7 +114,7 @@ const PERMIT_ALLOWED_TYPE = [
 ]
 
-export function useERC20Permit(
+function useERC20Permit(
```
many of these are removed probably because they dont use them and maybe it's more optimised? tho tree shaking should take care of that. will re-add them regardless tho

W3stside committed:
pr comments

W3stside committed:
cypress fix

W3stside committed:
remove projectId

W3stside committed:
fix overflow in managelist

W3stside committed:
style fix

W3stside committed:
style fixes

W3stside committed:
ci fix

W3stside commented on .github/workflows/ci.yml:
```diff
@@ -13,6 +13,7 @@ env:
   REPO_NAME_SLUG: gpswapui
   PR_NUMBER: ${{ github.event.number }}
   REACT_APP_SENTRY_DSN: ${{ secrets.SENTRY_DSN }}
+  CI: ""
```
we are checking prod for the issues there and if present we're fixing this later as it's not mega high prio

W3stside committed:
Merge remote-tracking branch 'origin/release/1.3.0' into uni-merge/4.13.0

W3stside committed:
remove CI: false from workflow

W3stside committed:
add new multicall xdai address (#1528)

W3stside committed:
set xdai polling to 1ms (#1529)

W3stside committed:
[Uni merge 🦄 ] replace uniswap text with cowswap (#1526)

* replace uniswap text with cowswap

* only show unsupported token tag

* comment unused types/imports

W3stside committed:
[Uni merge 🦄 ] Switch network adaptation (#1501)

* fix supported networks

* lock file

* style fixes

* edit menu/header to better responsive feel

* dead code in baseTheme

* expand mediaWidths in cowswapTheme

* remove styled base theme

* responsive header for network switcher

* Fix code style issues with Prettier

* fix network connect on no connection

* network switching UI/logic fix

* pr comment fix chainId

* fix disabled check

* styling

* style pt.2 - highlight in list selected network

* yarn update

* pr comments:

- network card style

* pr comments:

- text change

* pr comments

- move menu items into menu at medium

* pr comments

- deal better with unsupported networks

* non-connected - connect and switch

Co-authored-by: Lint Action <lint-action@samuelmeuli.com>

W3stside committed:
Merge branch 'release/1.3.0' into uni-merge/4.13.0

alfetopito reviewed (approved):
Not a through test, but working good for wrap/unwrap/swap

Only one thing to keep in mind before release, we need to remove `profile` page link again
![screenshot_2021-10-12_09-22-21](https://user-images.githubusercontent.com/43217/136994285-67b44019-2c1c-48c9-a755-515d501012de.png)

fairlighteth committed:
WIP Uni Merge - styles fixes. (#1565)

* Menu styles/icon fixes.

* Fix code style issues with Prettier

* Menu styles/icon fixes.

* Fix code style issues with Prettier

* Menu style fixes.

* iOS menu fix.

Co-authored-by: Lint Action <lint-action@samuelmeuli.com>
